File: FreelanceFlow/FreelanceFlow.Client/src/app/models/project.ts
export interface Project {
  id?: number;
  clientId: number;
  client?: { name: string; companyName: string };
  title: string;
  description: string;
  budget: number;
  status: 'Planning' | 'In Progress' | 'On Hold' | 'Completed';
  deadline: string;
  createdAt?: string;
}

export interface DashboardStats {
  clients: number;
  activeProjects: number;
  completedProjects: number;
  pipelineValue: number;
  invoiced: number;
  paid: number;
  trackedHours: number;
  upcomingDeadlines: { id: number; title: string; deadline: string; status: string }[];
}

File: FreelanceFlow/FreelanceFlow/Model/Project.cs
namespace FreelanceFlow.Model;

public class Project
{
    public long Id { get; set; }
    public long ClientId { get; set; }
    public Client? Client { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Budget { get; set; }
    public string Status { get; set; } = "Planning";
    public DateTime Deadline { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<Proposal> Proposals { get; set; } = new List<Proposal>();
}
